lib/semantics/semantics.cc

```diff
@@ -228,6 +230,35 @@ void DoDumpSymbols(std::ostream &os, const Scope &scope, int indent) {
   --indent;
 }
 
+void Semantics::DumpSymbolsSources(std::ostream &os) {
+  NameToSymbolMap symbols;
+  GetAllSymbols(context_.globalScope(), symbols);
+  for (const auto pair : symbols) {
```

Comment:
If you add `const Symbol &symbol{*pair.second};` here you can use `symbol` below and it will be easier to understand.